### PR TITLE
Issue #1877 Nodejs-ex gets stuck on rollout during integration test

### LINKED ISSUE #1877: Nodejs-ex gets stuck on rollout during integration test
### General information
As part of #1838 I tried to add a check to the http endpoint of nodejs-ex, but it gets stuck on rollout step. We need to do more investigation of where this issue is coming from. For now the http check was removed.

### Steps to reproduce

  1. Add rollout step to the nodejs-ex scenario in cmd-openshift.feature
  2. Add http check steps to the nodejs-ex scenario in cmd-openshift.feature

### Expected

The application rollouts successfully and its http endpoint is accessible.

### Actual

The tests get stuck on the rollout step and whole test run is then killed by a 3600s timeout.

### Logs

```
12:13:11   Scenario: User creates route to the service                                                                                                                                                             # features/cmd-openshift.feature:81
12:13:11     Given executing "oc set probe dc/nodejs-ex --readiness --get-url=http://:8080" succeeds                                                                                                               # integration_test.go:706 -> executingOcCommandSucceedsOrFails
13:30:32 Killed by signal 15.
13:30:32 + rtn_code=124
13:30:32 + '[' 124 -eq 0 ']'
```

### PR DESCRIPTION
**Pull request by agajdosi**
Fix #1877

### CODE REVIEW

**Comment by agajdosi** [comment]
retest this please

**Comment by budhrg** [comment]
retest this please

**Comment by budhrg** [comment]
retest this please

File: test/integration/features/cmd-openshift.feature [hunk 1]
```diff
@@ -35,13 +41,21 @@ cluster in VM provided by Minishift.
       Then stdout should contain "OpenShift restarted successfully"
        And stdout of command "minishift ssh -- "docker inspect --format={{.State.FinishedAt}} origin"" is not equal to "0001-01-01T00:00:00Z"
 
+  Scenario: registry check
+     Given executing "oc login -u system:admin" succeeds
+      When executing "oc project default" succeeds
+      When executing "oc get -o json pods" succeeds
+      Then stdout should contain "just to fail and print the result to console:"
```
Comment (praveenkumar) [inline test/integration/features/cmd-openshift.feature]: Why this scenario is repeated at diffferent place in this file?

Comment (agajdosi) [reply test/integration/features/cmd-openshift.feature]: @praveenkumar just to get information about docker-registry container. It seems that after restart of the OpenShift the container is not running: 
```
08:30:56                 "containerStatuses": [
08:30:56                     {
08:30:56                         "containerID": "docker://de3d6280c9d576441cca4dcadd897d3b17a0a71bae280507a23807c94635004a",
08:30:56                         "image": "openshift/origin-deployer:v3.7.1",
08:30:56                         "imageID": "docker-pullable://openshift/origin-deployer@sha256:064c3ba908fe9f7b4a22373bde98167789292de3670d48b05ef665748d1cadea",
08:30:56                         "lastState": {},
08:30:56                         "name": "deployment",
08:30:56                         "ready": false,
08:30:56                         "restartCount": 0,
08:30:56                         "state": {
08:30:56                             "terminated": {
08:30:56                                 "containerID": "docker://de3d6280c9d576441cca4dcadd897d3b17a0a71bae280507a23807c94635004a",
08:30:56                                 "exitCode": 1,
08:30:56                                 "finishedAt": "2018-03-14T08:30:33Z",
08:30:56                                 "reason": "Error",
08:30:56                                 "startedAt": "2018-03-14T08:30:16Z"
08:30:56                             }
08:30:56                         }
08:30:56                     }
```

**Comment by agajdosi** [comment]
So it looks like the docker-registry container is `terminated` due to an error after the restart. I am going to put a sleep after the restart to see whether it is not caused by deployment of an app soon after the restart which breaks the registry.

**Comment by gbraad** [comment]
any clues?

**Comment by budhrg** [comment]
@gbraad Some clue can be found here https://github.com/minishift/minishift/issues/1877#issuecomment-374955871.

File: Makefile [hunk 2]
```diff
@@ -214,9 +214,13 @@ integration_all: $(MINISHIFT_BINARY)
...
 
 .PHONY: integration_pr ## Run integration tests for pull request
-integration_pr: GODOG_OPTS = --tags=~coolstore\&\&~addon-xpaas
+integration_pr: GODOG_OPTS = --tags=cmd-openshift
 integration_pr: $(MINISHIFT_BINARY)
 	mkdir -p $(INTEGRATION_TEST_DIR)
+	echo "HTTP_PROXY=$(HTTP_PROXY) $(origin HTTP_PROXY)"
+	echo "HTTPS_PROXY=$(HTTPS_PROXY) $(origin HTTPS_PROXY)"
+	echo "MINISHIFT_HTTP_PROXY=$(MINISHIFT_HTTP_PROXY) $(origin MINISHIFT_HTTP_PROXY)"
+	echo "MINISHIFT_HTTPS_PROXY=$(MINISHIFT_HTTPS_PROXY) $(origin MINISHIFT_HTTPS_PROXY)"
```
Comment (praveenkumar) [inline Makefile]: @agajdosi what all those echoes for?

Comment (agajdosi) [reply Makefile]: @praveenkumar oh, I have missed those, was just for debugging purposes, going to remove it now. thanks for noticing.

Comment (agajdosi) [reply Makefile]: also line 217 needs fix

File: test/integration/features/cmd-openshift.feature [hunk 3]
```diff
@@ -31,6 +31,7 @@ cluster in VM provided by Minishift.
         On restart of OpenShift cluster containers are terminated, which sets FinishedAt to actual time. This value persist
         after next start of container.
      Given stdout of command "minishift ssh -- "docker inspect --format={{.State.FinishedAt}} origin"" is equal to "0001-01-01T00:00:00Z"
+       And user waits "10" seconds #TO BE REMOVED with OpenShift v3.9. OpenShift v3.7 needs additional time after startup to get containers ready due to #2169.
```
Comment (LalatenduMohanty) [inline test/integration/features/cmd-openshift.feature]: @agajdosi Why the sleep is before the restart? I thought we need sleep after restart?

Comment (agajdosi) [reply test/integration/features/cmd-openshift.feature]: @LalatenduMohanty I was suspecting the same but later I have found out the restart so soon after startup causes troubles. Issue: https://github.com/minishift/minishift/issues/2169

Comment (LalatenduMohanty) [reply test/integration/features/cmd-openshift.feature]: Also we have moved to v3.9.0 recently. So we should try running it without sleep.

File: test/integration/features/cmd-openshift.feature [hunk 4]
```diff
@@ -31,6 +31,8 @@ cluster in VM provided by Minishift.
         On restart of OpenShift cluster containers are terminated, which sets FinishedAt to actual time. This value persist
         after next start of container.
      Given stdout of command "minishift ssh -- "docker inspect --format={{.State.FinishedAt}} origin"" is equal to "0001-01-01T00:00:00Z"
+#TO BE REMOVED with OpenShift v3.9. OpenShift v3.7 needs additional time after startup to get containers ready due to #2169.
```
Comment (budhrg) [inline test/integration/features/cmd-openshift.feature]: @agajdosi Now, 3.9.0 is default openshift version. Would you consider removing comment now?

**Comment by budhrg** [comment]
retest this please

**Pull request merged**